packages/uiweb/src/lib/components/chat/ChatPreviewList/ChatPreviewList.tsx

```diff
@@ -258,6 +257,27 @@ export const ChatPreviewList: React.FC<IChatPreviewListProps> = (
           errored: false,
           error: null,
         }));
+        if (
+          listInnerRef &&
+          listInnerRef?.current &&
+          listInnerRef?.current?.parentElement
+        ) {
+       
+          if (
+            listInnerRef.current.clientHeight + 25 >
+            listInnerRef.current.parentElement.clientHeight
+          ) {
+            // set loading to true
+            setChatPreviewList((prev) => ({
+              ...prev,
+              nonce: generateRandomNonce(),
+              loading: true,
+            }));
+          }
+        }
```

Comment:
Ideal placement of this will be in useEffect, when the items change, at that time, we need to check. It should only run when if preloading is not true, ie, the page has not been loaded

Reply:
Also have onscroll as a constant and use that instead of 25 as a fixed value, update onscroll as well.

Reply:
okay